### DISCUSSION TITLE
Sliding window forecast

### DISCUSSION THREAD

**Discussion by nazrifuad2020** [Nixtla/mlforecast#172] [Q&A]
Hi, I am considering using mlforecast for my forecasting project that uses sliding window approach. I have a question regarding the correct way to do this without model refitting. Based on the end-to-end walkthrough, I understand that once you have made prediction and new target data is available, you can incorporate this new data using MLForecast.ts.update. However, I discovered that for subsequent new prediction, instead of using the previously incorporated target data (lag1) and the target data that came before it for subsequent lags (my model uses consecutive lags=[1,2,3,...]), the model will use the previous forecasts for subsequent lags. Is my current approach wrong? What is the correct way to do this?

**Comment by jmoralez** [comment]
Hi. Are you using v0.8.0?

**Reply by nazrifuad2020** [reply]
As of today, I have started using v0.8.1 and from what I can see now, with regards to the issue above, it has produced the behavior that is now expected. In fact when I compared with another forecasting package, using same sklearn model and forecasting settings, both now produced the same results. So kudos to all developers here. You may consider this issue closed. Thank you.

**Reply by jmoralez** [reply]
Awesome! I introduced this bug yesterday, sorry about that. Your report made me realize it and fix it, so thank you!
